packages/codeeditor/src/tokens.ts

```diff
@@ -30,3 +32,7 @@ export interface IEditorServices {
    */
   readonly mimeTypeService: IEditorMimeTypeService;
 }
+
+export interface IEditorServices2 {
```

Comment:
was this intentional?

Reply:
Oops!

Reply:
Ahhh yes, this was the thing I wanted to ask about. The factories are currently constants, so they should probably be methods that take the optional translator so we can propagate them down the factory.

Thoughts?

```typescript
/**
 * Code editor services.
 */
export interface IEditorServices {
  /**
   * The code editor factory.
   */
  readonly factoryService: IEditorFactoryService;

  /**
   * The editor mime type service.
   */
  readonly mimeTypeService: IEditorMimeTypeService;
}
```

to

```typescript

/**
 * Code editor services.
 */
export interface IEditorServices {
  /**
   * The code editor factory.
   */
  readonly getFactoryService(translator?: ITranslator): IEditorFactoryService;

  /**
   * The editor mime type service.
   */
  readonly getMimeTypeService(translator?: ITranslator): IEditorMimeTypeService;
}
```

Reply:
Yep, sounds reasonabl